modules/core/src/main/scala/com.snowplowanalytics.snowplow.lakes/processing/SparkUtils.scala

```diff
@@ -59,44 +58,48 @@ private[processing] object SparkUtils {
     writer.sparkConfig ++ config.conf + (gcpUserAgentKey -> gcpUserAgentValue)
   }
 
-  def saveDataFrameToDisk[F[_]: Sync](
+  def initializeLocalDataFrame[F[_]: Sync](spark: SparkSession, viewName: String): F[Unit] =
+    for {
+      _ <- Logger[F].debug(s"Initializing local DataFrame with name $viewName")
+      _ <- Sync[F].blocking {
+             spark.emptyDataFrame.createTempView(viewName)
+           }
+    } yield ()
+
+  def localAppendRows[F[_]: Sync](
     spark: SparkSession,
+    viewName: String,
     rows: NonEmptyList[Row],
     schema: StructType
-  ): F[DataFrameOnDisk] = {
-    val count = rows.size
+  ): F[Unit] =
     for {
-      viewName <- Sync[F].delay("v" + UUID.randomUUID.toString.replaceAll("-", ""))
-      _ <- Logger[F].debug(s"Saving batch of $count events to local disk")
+      _ <- Logger[F].debug(s"Saving batch of ${rows.size} events to local DataFrame $viewName")
       _ <- Sync[F].blocking {
              spark
                .createDataFrame(rows.toList.asJava, schema)
                .coalesce(1)
                .localCheckpoint()
-               .createTempView(viewName)
+               .unionByName(spark.table(viewName), allowMissingColumns = true)
+               .createOrReplaceTempView(viewName)
```

Comment:
Curious: why `createOrReplaceTempView` here if we do that already in `initializeLocalDataFrame`?

Reply:
Think of this as `.replaceTempView` (but Spark doesn't have a method called that).

Data Frames are immutable.  So the loader runs like this:

- Create an empty DataFrame and save it under the name `v123`
- Later, create a new DataFrame by union'ing `v123` with some new data.  Save that new DataFrame under the name `v123`
- Same again
- etc